service/storage_service.cc

```diff
@@ -5425,22 +5420,22 @@ void storage_service::init_messaging_service(sharded<service::storage_proxy>& pr
                 // To fix that, we could store in topology a small history of recent CDC generation IDs
                 // (garbage-collected with time) instead of just the last one, and load all of them.
                 // Alternatively, a node would wait for some time before switching to normal state.
-                auto s = ss._db.local().find_schema(db::system_keyspace::NAME, db::system_keyspace::CDC_GENERATIONS_V3);
-                auto key = dht::decorate_key(*s, partition_key::from_singular(*s, curr_cdc_gen_id->id));
-                auto partition_range = dht::partition_range::make_singular(key);
+                //
+                // For now, this problem is temporarily solved by sending the entire CDC_GENERATIONS_V3 table.
```

Comment:
Not exactly -- we are still only fetching `current_cdc_generation` from the table into memory on nodes that apply topology commands or group 0 snapshot.

You did improve the situation with your PR -- the nodes are now **receiving** all generations, however, they still only looking at the `current` one.

So you could improve the comment above: instead of "The last bootstrapped node will only receive the newest CDC generation...", write something like "The last bootstrapped node will only read the newest CDC generation into memory...".